templates/repo/diff/box.tmpl

```diff
@@ -31,12 +31,57 @@
 			{{end}}
 			{{template "repo/diff/whitespace_dropdown" .}}
 			{{template "repo/diff/options_dropdown" .}}
+			{{if .PageIsPullFiles}}
+				<script id="commit-data-script">
+					{{$numCommits := len .Commits}}
+					window.config.pageData.commitInfo = {
+						issueLink: {{$.Issue.Link}},
+						lastReviewCommitSha: {{.LastReviewSha}},
+						commits: [
+						{{range .Commits}}
+								{
+									Summary: {{.Summary}},
+									CommitterOrAuthorName: {{if .Committer}}{{.Committer.Name}}{{else}}{{.Author.Name}}{{end}},
+									ID: {{.ID.String}},
+									Time: {{if .Committer}}{{.Committer.When}}{{else}}{{.Author.When}}{{end}},
+									Selected: false
+								},
+						{{end}}],
+						locale: {
+							lang: {{$.locale}},
+							filter_changes_by_commit: {{.locale.Tr "repo.pulls.filter_changes_by_commit"}},
+							show_all_commits: {{.locale.Tr "repo.pulls.show_all_commits"}},
+							stats_num_commits: {{.locale.TrN $numCommits "repo.activity.git_stats_commit_1" "repo.activity.git_stats_commit_n" $numCommits}},
+							show_changes_since_your_last_review: {{.locale.Tr "repo.pulls.show_changes_since_your_last_review"}},
+						},
+						queryParams: '?style={{if $.IsSplitStyle}}split{{else}}unified{{end}}&whitespace={{$.WhitespaceBehavior}}&show-outdated={{$.ShowOutdatedComments}}'
+					};
```

Comment:
@wxiaoguang is this correct usage of `window.config.pageData`? I'm not sure myself.

Reply:
I do not think it's good to put everything in one page request.

Ideally, the "popup" element can do lazy-loading. Use a separate backend handler to provide related data, the frontend code just does "fetch"

Reply:
IIRC, fomantic popup has some way of lazy-loading, but I guess we should not use it because we eventually want to migrate away from fomantic, and this would just problematically deepen our integration against it.

Reply:
This PR's code is not related to Fomantic's JS.

Reply:
No, but how would you implement "popup" element can do lazy-loading"?

Reply:
> Just `fetch` before opening it?

Yes, that's a general approach, simple and clear enough.

Reply:
yes i understand that only loading the data when needed is a good approach. But in this case all of the information where already available on the page, so i was just using the information in viewPullfiles.

Reply:
If the template block is only loaded where needed and the data is already computed, I see no issue doing it like this because it will be better UX if they don't have to wait for the `fetch`.

Reply:
Not sure - if you want i can create a new endpoint and fetch the information when opening the component.

What do you think @wxiaoguang ?

Reply:
If all related data is already prepared by the existing backend code, then I am neutral for it (current approach or fetch/lazy-loading)

A pure fetch solution could avoid the fragile `pageData`, but it's not a must at the moment IMO.

Reply:
i just added a new route and the component now fetches the data.